bazel/protobuf.patch

```diff
@@ -87,3 +87,80 @@
      # python_message.py registers all Message classes to some pickle global
      # registry, which makes the classes immortal.
      # We save a copy of this registry, and reset it before we could references.
+
+--- toolchain/BUILD
++++ toolchain/BUILD
+@@ -14,6 +14,7 @@ TOOLCHAINS = {
+     "linux-x86_64": "cc-compiler-linux-x86_64",
+     "win32": "cc-compiler-windows-x86_32",
+     "win64": "cc-compiler-windows-x86_64",
++    "k8": "cc-compiler-k8",
+ }
+
+ cc_toolchain_suite(
+@@ -39,6 +40,14 @@ cc_toolchain_suite(
+     for cpu, toolchain in TOOLCHAINS.items()
+ ]
+ 
++cc_toolchain_config(
++    name = "k8-config",
++    linker_path = "/usr/bin/ld",
++    sysroot = "/opt/manylinux/2014/x86_64",
++    target_cpu = "x86_64",
++    target_full_name = "x86_64-linux-gnu",
++)
++
+ cc_toolchain_config(
+     name = "linux-aarch_64-config",
+     sysroot = "/opt/manylinux/2014/aarch64",
+@@ -85,6 +85,7 @@ cc_toolchain_config(
+         "-I/usr/tools/apple_sdks/xcode_13_0/macosx/usr/include/c++/v1",
+         "-I/usr/tools/apple_sdks/xcode_13_0/macosx/usr/include"
+     ],
++    extra_linker_flags = ["-undefined", "dynamic_lookup"],
```

Comment:
`-undefined dynamic_lookup` should go directly onto the Python `cc_binary(linkshared=1)` rule, not on the toolchain.

We don't want `-undefined dynamic_lookup` be used on regular binaries. Regular binaries should throw link errors when a symbol is not available, otherwise the binary will fail at runtime:

Good behavior, this is what we want:
```
$ cat foo.c
int f();

int main() {
  return f();
}
$ gcc -o foo foo.c
Undefined symbols for architecture x86_64:
  "_f", referenced from:
      _main in foo-db1b0c.o
ld: symbol(s) not found for architecture x86_64
```

Bad behavior, the link succeeds but the binary fails at runtime.
```
$ gcc -o foo foo.c -undefined dynamic_lookup
(base) √ Joshs-MacBook-Pro ~ $ ./foo 
dyld[46235]: missing symbol called
zsh: abort      ./foo
```

Reply:
Moved!